### PR TITLE
Make company-idle-delay support predicate function

### PR DESCRIPTION
**Pull request by amosbird**
Sample usage:

`(setq company-idle-delay (lambda () (if (company-in-string-or-comment) nil 0.3)))`

### CODE REVIEW

**Comment by dgutov** [comment]
Thanks!

Is this your first contribution to Emacs or GNU ELPA projects?

**Comment by amosbird** [comment]
> Is this your first contribution to Emacs or GNU ELPA projects?

Yes. Do I need to sign some paper first?

**Comment by dgutov** [comment]
Not yet. You might want to do that before the next time, though. Let me know if you'd like the instructions.

File: company.el [hunk 1]
```diff
@@ -1702,25 +1703,31 @@ prefix match (same case) will be prioritized."
...
-                 (company--should-begin)
-                 (setq company-timer
-                       (run-with-timer (company--idle-delay) nil
-                                       'company-idle-begin
-                                       (current-buffer) (selected-window)
-                                       (buffer-chars-modified-tick) (point))))))
+            (let ((delay (company--idle-delay)))
+             (and (or (numberp delay)
+                      ;; Deprecated.
+                      (eq delay t))
```
Comment (dgutov) [inline company.el]: I think there's no need to handle the value `t` here anymore.

Comment (amosbird) [reply company.el]: Done.

**Comment by amosbird** [comment]
> Let me know if you'd like the instructions.
Yes pls. I've tried reaching copyright-clerk@fsf.org but not getting any responds yet.

**Comment by dgutov** [comment]
The copyright clerk is fairly busy, so maybe wait a little while. Here are the instructions, though:

https://code.orgmode.org/bzg/org-mode/raw/master/request-assign-future.txt

(Keep the first answer as is).

**Comment by dgutov** [comment]
Could you also add a NEWS entry? At the top.

**Comment by amosbird** [comment]
NEWS added. 

 > Please use your full legal name (in ASCII characters)

By legal name it means `Amos Bird` ?

**Comment by dgutov** [comment]
> By legal name it means `Amos Bird` ?

Probably. Whatever is written in your ID or international passport.

**Pull request merged**